File: implementations/node/sdk/src/contracts/control/execution-cancellation-request.ts
import { AI_SDK_SCHEMA_VERSIONS } from "../common/schema-versions.js";

export interface AiSdkExecutionCancellationRequest {
  readonly schemaVersion?: typeof AI_SDK_SCHEMA_VERSIONS.executionCancellationRequest;
  readonly reason?: string;
  readonly correlationId?: string;
}

File: implementations/python/sdk/src/multiplexed_ai_sdk/contracts/control/execution_cancellation_request.py
from dataclasses import dataclass

from ...wire_model import AiSdkWireModel
from ..common.schema_versions import AiSdkSchemaVersions


@dataclass(frozen=True)
class AiSdkExecutionCancellationRequest(AiSdkWireModel):
    schema_version: int = AiSdkSchemaVersions.EXECUTION_CANCELLATION_REQUEST
    reason: str | None = None
    correlation_id: str | None = None
